### DISCUSSION TITLE
Node Validation

### DISCUSSION THREAD

**Discussion by oh-tarnished** [xyflow/xyflow#2256] [Q&A]
Hey People, Was wondering if there was any kind of node validation that can be done on editor while connecting. Also is there a way to delete an edge? Please help  thanks.

**Comment by bcakmakoglu** [comment]
In custom node components with `<Handle>` you can use the `isValidConnection` prop.
[Read more about it in the docs.](https://reactflow.dev/docs/api/nodes/handle/#prop-types)

**Reply by bcakmakoglu** [reply]
[There's also an example here.](https://reactflow.dev/docs/examples/interaction/validation/)

**Reply by oh-tarnished** [reply]
Hey @bcakmakoglu thanks for the quick reply, can we add 20 inputs to a single node ? thank you again.

**Reply by bcakmakoglu** [reply]
You mean the small dots (described as `Handle`) that connect edges?
Technically you can add as many as you want, you just have to make sure to call `updateNodeInternals` when you dynamically add / remove them, otherwise your node does not know about the handles and edges will not be properly aligned.
`useUpdateNodeInternals` can be found on the `reactFlowInstance`.
[More on that here.](https://reactflow.dev/docs/api/react-flow-provider/)

**Reply by oh-tarnished** [reply]
Thanks you @bcakmakoglu a lot for helping me out. 😀
